test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SearchTestsBase.cs

```diff
@@ -96,12 +120,25 @@ protected async Task<Bundle> ExecuteAndValidateBundle(string searchUrl, bool sor
             return firstBundle;
         }
 
+        protected async Task<OperationOutcome> ExecuteAndValidateErrorOperationOutcomeAsync(
+            string searchUrl,
+            Tuple<string, string> customHeader,
+            HttpStatusCode expectedStatusCode,
+            OperationOutcome expectedOperationOutcome)
+        {
+            FhirException ex = await Assert.ThrowsAsync<FhirException>(() => Client.SearchAsync(searchUrl, customHeader));
+            Assert.Equal(expectedStatusCode, ex.StatusCode);
+            ex.OperationOutcome.Id = null;
+            Assert.True(expectedOperationOutcome.IsExactly(ex.OperationOutcome), "Deep compare detected expected and actual OperationOutcome mismatch.");
+            return ex.OperationOutcome;
```

Comment:
Nit: since this method is only called once, could we keep this logic inside the unit test? Probably a personal preference, but I think it might also make the test a bit easier to read (happy to change my mind if you disagree)

Reply:
I'm just following convention. All unit tests in this file use "ExecuteAndValidate" functions. I'm probably first one in this file to validate proper error OperationOutcome messages. I would rather add few more unit tests and call this function than put this inside the unit test. I can see that same function could be used for search test failures, too many sort parameters, mix of search and sort failures. I can do that instead.

Reply:
Added unit tests for case when sort is not valid search param. Now method is reused.